docs/aws/audit/iamcompliance/rules/root_account_access_keys.mdx

```diff
@@ -23,6 +23,214 @@ HIPAA, ISO27001, PCIDSS, GDPR, NIST, SOC2, CISAWS, CBP, HITRUST, AWSWAF, NISTCSF
+   ```sh
+   aws iam enable-mfa-device --user-name root --serial-number arn:aws:iam::123456789012:mfa/root-account-mfa-device --authentication-code1 123456 --authentication-code2 789012
+   ```
+
+By following these steps, you can ensure that the root account does not have access keys, and you can use an IAM user with administrative privileges for necessary tasks.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the root account from having access keys in IAM using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with the necessary credentials and region information.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   ```
+
+3. **Check for Root Access Keys**:
+   Use the IAM client to list access keys for the root account and ensure no access keys exist.
+   ```python
+   iam_client = session.client('iam')
+
+   # List access keys for the root account
+   response = iam_client.list_access_keys(UserName='root')
+
+   # Check if any access keys exist
+   if response['AccessKeyMetadata']:
+       print("Root account has access keys. Please remove them.")
+   else:
+       print("Root account does not have any access keys.")
+   ```
+
+4. **Automate the Check and Preventive Action**:
+   You can automate this check to run periodically and alert or take action if access keys are found.
+   ```python
+   import boto3
+   from botocore.exceptions import NoCredentialsError, PartialCredentialsError
+
+   def check_root_access_keys():
+       try:
+           session = boto3.Session(
+               aws_access_key_id='YOUR_ACCESS_KEY',
+               aws_secret_access_key='YOUR_SECRET_KEY',
+               region_name='YOUR_REGION'
+           )
+           iam_client = session.client('iam')
+
+           # List access keys for the root account
+           response = iam_client.list_access_keys(UserName='root')
+
+           # Check if any access keys exist
+           if response['AccessKeyMetadata']:
+               print("Root account has access keys. Please remove them.")
+               # Optionally, you can add code here to delete the keys automatically
+               # for key in response['AccessKeyMetadata']:
+               #     iam_client.delete_access_key(UserName='root', AccessKeyId=key['AccessKeyId'])
+           else:
+               print("Root account does not have any access keys.")
+       except (NoCredentialsError, PartialCredentialsError) as e:
+           print(f"Error: {e}")
+
+   # Run the check
+   check_root_access_keys()
+   ```
+
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>